File: common/src/main/java/dev/number6/slack/port/HttpPort.java
package dev.number6.slack.port;

import dev.number6.slack.CallResponse;
import com.amazonaws.services.lambda.runtime.LambdaLogger;

public interface HttpPort {

//    CallResponse makeHttpCall(T request);

    CallResponse get(String url, LambdaLogger logger);

    CallResponse post(String url, String body, LambdaLogger logger);
}

File: slack/src/main/kotlin/dev/number6/slack/port/HttpPort.kt
package dev.number6.slack.port

import com.amazonaws.services.lambda.runtime.LambdaLogger
import dev.number6.slack.model.CallResponse

internal interface HttpPort {
    fun get(url: String, logger: LambdaLogger): CallResponse
    fun post(url: String, body: String, logger: LambdaLogger): CallResponse
}